Program.cs

```diff
@@ -18,42 +17,132 @@ public int add(string Numbers)
 
             try
             {
-                return tryAdddelimiters(Numbers);
+                return tryAdddelimitersWithNegitaveIgnoreingBigNumbers(Numbers);
             }
-            catch (System.Exception)
+
+            catch (FormatException)
             {
                 return 0;
             }
 
         }
-        private int tryAdd(string Numbers)
+        private int tryAddSimple(string Numbers)
         {
             int ans;
             ans = Numbers.Where(x => x != ',').Aggregate((sum, i) => sum += i);
             ans -= Numbers.Where(x => x != ',').Count() * '0';
             return ans;
         }
+        private int tryAddbig(string Numbers)
+        {
+            List<string> Nums = Numbers.Split(',').ToList();
+            int sum = GetSum(Nums);
+            return sum;
+        }
         private int tryAddWithEndLine(string Numbers)
         {
-            int ans;
-            string endLine = "/n";
-            Numbers = Numbers.Replace("/n", "");
-            ans = Numbers.Where(x => x != ',').Aggregate((sum, i) => sum += i);
-            ans -= Numbers.Where(x => x != ',').Count() * '0';
-            return ans;
+            Numbers = Numbers.Replace("/n", ",");
+            List<string> Nums = Numbers.Split(',').ToList();
+            int sum = GetSum(Nums);
+            return sum;
         }
-         private int tryAdddelimiters(string Numbers)
+        private int tryAdddelimiters(string Numbers)
         {
-            int ans;
-            char endLine = Numbers[2];
-            Numbers = Numbers.Replace("/n", "");
-            Numbers = Numbers.Replace("//", "");
-            ans = Numbers.Where(x => x != endLine).Aggregate((sum, i) => sum += i);
-            ans -= Numbers.Where(x => x != endLine).Count() * '0';
-            return ans;
+            char diameter = ',';
+            if (Numbers.StartsWith("//"))
+            {
+                diameter = Getdiameter(Numbers);
+            }
+            List<string> Nums = Numbers.Split(diameter).ToList();
```

Comment:
the requirement says all old cases should still be supported 
I don't think the comma case is still supported here.